std/neko/_std/String.hx

```diff
@@ -71,7 +71,14 @@ import haxe.iterators.StringKeyValueIterator;
 
 	public function indexOf( str : String, ?startIndex : Int ) : Int {
 		untyped {
-			var p = try __dollar__sfind(this.__s,if( startIndex == null ) 0 else startIndex,str.__s) catch( e : Dynamic ) null;
+			var l = __dollar__ssize(this.__s);
+			if( startIndex == null || startIndex < -l )
+				startIndex = 0;
+			if( startIndex > l )
```

Comment:
This should be `>=` (which also means there's a test missing).

Reply:
Also this can be `else if` because it cannot be entered if the previous branch was.

Reply:
I'm not sure about that.

I think `"foobar".indexOf("", 6)` should return `6` (well, consistent with js and java https://github.com/HaxeFoundation/haxe/issues/7609#issuecomment-480910920). If the string is not empty the actual call to `$sfind` throws, so `-1` is returned.

Reply:
You're right. I thought it should return the offset of the last character, but that's not correct.